lib/litani.py

```diff
@@ -147,44 +190,10 @@ async def try_acquire_wait(self, timeout=0):
 
 
 
-def _get_cache_dir(path=os.getcwd()):
-    def cache_pointer_dirs():
-        current = pathlib.Path(path).resolve(strict=True)
-        yield current
-        while current.parent != current:
-            current = current.parent
-            yield current
-
-    for possible_dir in cache_pointer_dirs():
-        logging.debug(
-            "Searching for cache pointer in %s", possible_dir)
-        possible_pointer = possible_dir / CACHE_POINTER
-        try:
-            if possible_pointer.exists():
-                logging.debug(
-                    "Found a cache pointer at %s", possible_pointer)
-                with open(possible_pointer) as handle:
-                    pointer = handle.read().strip()
-                possible_cache = pathlib.Path(pointer)
-                if possible_cache.exists():
-                    logging.debug("cache is at %s", possible_cache)
-                    return possible_cache
-                logging.warning(
-                    "Found a cache file at %s pointing to %s, but that "
-                    "directory does not exist. Continuing search...",
-                    possible_pointer, possible_cache)
-        except PermissionError:
-            pass
-
-    logging.error(
-        "Could not find a pointer to a litani cache. Did you forget "
-        "to run `litani init`?")
-    raise FileNotFoundError
-
-
 def get_cache_dir(path=os.getcwd()):
     try:
-        return _get_cache_dir(path)
+        cache_dir = CacheDir(lookup_path=path)
```

Comment:
Just as a matter of personal preference I think the construction of this object ought to go outside of the try/except because you shouldn't expect that constructing an object will raise an exception.

Reply:
Also, is there a reason that we pass the lookup_path into the constructor rather than just to the `get()` method?

Reply:
> Just as a matter of personal preference I think the construction of this object ought to go outside of the try/except because you shouldn't expect that constructing an object will raise an exception.

Makes sense. Will update.


> Also, is there a reason that we pass the lookup_path into the constructor rather than just to the `get()` method?

Because previously we were passing a `path` var to the `_get_cache_dir` function which was set to `os.getcwd()` as default.

Reply:
I don't understand. Can we not now pass a `path` var into `get()` that is set to `os.getcwd()` as default? Why does it need to go in the constructor?

Reply:
Resolved